src/services/functions.js

```
@@ -0,0 +1,301 @@
+const { Pool } = require('pg');
+const jwt = require('jsonwebtoken');
+
+const { jsonResponse, streamToString } = require('../modules/utils');
+const { hash, verify } = require('../modules/passwordHash');
+const { db } = require('../config/config.js');
+
+const pool = new Pool(db);
+
+const queries = {
+  updateJson: `
+  SELECT posts.title, posts.text, posts.price, posts.code, posts.link, posts.category_id, media.src
+  FROM posts
+  INNER JOIN media
+  ON posts."ID" = media.post_id
+  WHERE post_type_id = '1';`,
+};
+
+function generateAccessToken(payload) {
+  return jwt.sign(payload, process.env.ACCESS_TOKEN_SECRET, {
+    expiresIn: '10m',
+  });
+}
+
+function generateRefreshToken(payload) {
+  return jwt.sign(payload, process.env.REFRESH_TOKEN_SECRET, {
+    expiresIn: '90d',
+  });
+}
+
+async function verifyAccessToken(req) {
+  let verifiedPayload;
+  const bearerHeader = req.headers.authorization;
+  if (!bearerHeader) {
+    return null;
+  }
+  const bearer = bearerHeader.split(' ');
+  const accessToken = bearer[1];
+  jwt.verify(accessToken, process.env.ACCESS_TOKEN_SECRET, (err, payload) => {
+    if (err) return null;
+    verifiedPayload = payload;

```

It's not ok to ignore error a such way and extracting variable from callback to outer context. You need here `new Promise((resolve, reject) => .... verify(....., (err, payload) => {  if (err) reject(err) else resolve(payload) })....`